builder/frameworks/component_manager.py

```
@@ -0,0 +1,589 @@
+        # Create backup before first component removal
+        if remove_components and not self.removed_components:
+            self._backup_pioarduino_build_py()
+        
+        component_yml_path = self._get_or_create_component_yml()
+        component_data = self._load_component_yml(component_yml_path)
+        
+        if remove_components:
+            try:
+                components_to_remove = self.env.GetProjectOption("custom_component_remove").splitlines()
+                self._remove_components(component_data, components_to_remove)
+            except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)